docs/src/main/asciidoc/datasource-guide.adoc

```diff
@@ -0,0 +1,212 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+
+include::./attributes.adoc[]
+= {project-name} - Datasources
+
+Many projects that use data require connections to a database.
+The main way of obtaining connections to a database is to use a datasource.
+In {project-name}, the out of the box datasource implementation is Agroal.
+This guide will explain how to:
+
+* configure a datasource, or multiple datasources
+* how to obtain a reference to those datasources in code
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* less than 10 minutes
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven 3.5.3+
+
+== Creating the Maven project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=agroal-datasources\
+    -DclassName="org.acme.events.GreetingResource" \
+    -Dpath="/hello"
+----
+
+It generates:
+
+* the Maven structure
+* a landing page accessible on `http://localhost:8080`
+* example `Dockerfile` files for both `native` and `jvm` modes
+* the application configuration file
+* an `org.acme.events.GreetingResource` resource
+* an associated test
+
+== Adding maven dependencies
+
+Next, you will need to add the `quarkus-agroal` dependency to your project.
+There are a couple of ways to do this.
+You can add it explicitly, or transitively by adding the `quarkus-hibernate-orm` dependency.
+
+If you want to add the dependency explicitly:
+
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-agroal</artifactId>
+</dependency>
+--
+
+If you'd like to add Hibernate ORM instead, and get Agroal transitively:
+
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-hibernate-orm</artifactId>
+</dependency>
+--
+
+You will also need to add the database connector library of choice.
+Quarkus supports
+
+* H2
+* MariaDB (and MySQL)
+* Postgresql
+
+Do so by adding one of these dependencies to your pom.xml file:
+
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-h2</artifactId>
+</dependency>
+--
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-mariadb</artifactId>
+</dependency>
+--
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-postgresql</artifactId>
+</dependency>
+--
+
+== Configuring the datasource
+
+After all of the dependencies are added to your pom.xml file, you'll need to configure Agroal.
```

Comment:
I would just say `the dependencies` instead of `all of the...` because you don't want to add all the dependencies we talked about earlier.